src/Modules/ASayModule/Templates/Announcement.mt

```diff
@@ -1,49 +1,88 @@
+                    substyle="BgFrame1"
+                    halign="left" 
+                    colorize="{{ Theme.ASayModule_Announcement_Default_BgSecondary }}" 
+                    opacity="1"
+                    pos="0 0"
+                    valign="center"
+                    z-index="2" />
         </frame>
-        <script main="true">
-            <!--
-            #Include "MathLib" as ML
-            #Const C_MaxWidth 160.
-            #Const C_IconWidth 12.
-            
-            main() {
-                declare CMlFrame Frame <=> Page.GetFirstChild("window") as CMlFrame;
-								declare CMlFrame InnerFrame <=> Page.GetFirstChild("textFrame") as CMlFrame;
-                declare CMlLabel Label = Page.GetFirstChild("text") as CMlLabel;
-                declare CMlQuad BgQuad <=> Page.GetFirstChild("bgQuad") as CMlQuad;
-                declare Real Width = Label.ComputeWidth(Label.Value);
-                declare CMlLabel DbgLabel <=> Page.GetFirstChild("debugStuff") as CMlLabel;
-                if (Width >= C_MaxWidth) {
-                    Width = C_MaxWidth;
-                }
-								Width += C_IconWidth;
-                Frame.RelativePosition_V3.X = -(Width) /2.;
-                BgQuad.Size.X = Width;
-                InnerFrame.RelativePosition_V3.X = 11.;
-                if (Width == (C_MaxWidth + C_IconWidth)) {
-                    declare Real Pos = Width/2. +5;
-                    Label.RelativePosition_V3.X = Pos;
-                    while(True) {
-                            yield;
-                            Label.RelativePosition_V3 = <(ML::Sin(Now*0.0005)) * (Width/8), 0.>;	
-                    }
-                }
-            }
-            
-            -->
-        </script>
     </template>
+
+    <script main="true">
+      <!--
+      #Include "MathLib" as ML
+      #Const C_MaxWidth 160.
+      #Const C_IconWidth 12.
+      
+      main() {
+          declare CMlFrame Frame <=> Page.GetFirstChild("window") as CMlFrame;
+                          declare CMlFrame InnerFrame <=> Page.GetFirstChild("textFrame") as CMlFrame;
+          declare CMlLabel Label = Page.GetFirstChild("text") as CMlLabel;
+          declare CMlQuad BgQuad <=> Page.GetFirstChild("bgQuad") as CMlQuad;
+          declare Real Width = Label.ComputeWidth(Label.Value);
+          declare CMlLabel DbgLabel <=> Page.GetFirstChild("debugStuff") as CMlLabel;
+          if (Width >= C_MaxWidth) {
+              Width = C_MaxWidth;
+          }
+                          Width += C_IconWidth;
+          Frame.RelativePosition_V3.X = -(Width) /2.;
+          BgQuad.Size.X = Width;
+          InnerFrame.RelativePosition_V3.X = 11.;
+          if (Width == (C_MaxWidth + C_IconWidth)) {
+              declare Real Pos = Width/2. +5;
+              Label.RelativePosition_V3.X = Pos;
+              while(True) {
+                      yield;
+                      Label.RelativePosition_V3 = <(ML::Sin(Now*0.0005)) * (Width/8), 0.>;	
```

Comment:
Another possibility would be to use `AnimMgr.Add` and a sleep timeout to alternate between positions. 
I don't know if that would have any performance benefits for the client, but maybe AnimMgr can move UI elements more efficiently than pure maniascript.

Reply:
Made an issue for it: https://github.com/EvoEsports/EvoSC-sharp/issues/235